src/impl_owned_array.rs

```diff
@@ -1,5 +1,41 @@
 use imp_prelude::*;
 
+/// Methods specific to `Array0`.
+///
+/// ***See also all methods for [`ArrayBase`]***
+///
+/// [`ArrayBase`]: struct.ArrayBase.html
+impl<A> Array<A, Ix0> {
+    /// Returns the single element in the array without cloning it.
+    ///
+    /// ```
+    /// use ndarray::{arr0, Array0};
+    ///
+    /// // `Foo` doesn't implement `Clone`.
+    /// #[derive(Debug, Eq, PartialEq)]
+    /// struct Foo;
+    ///
+    /// let array: Array0<Foo> = arr0(Foo);
+    /// let scalar: Foo = array.into_scalar();
+    /// assert_eq!(scalar, Foo);
+    /// ```
+    pub fn into_scalar(mut self) -> A {
+        let size = ::std::mem::size_of::<A>();
+        if size == 0 {
+            // Any index in the `Vec` is fine since all elements are identical.
+            self.data.0.remove(0)
+        } else {
+            // Find index in the `Vec` corresponding to `self.ptr`.
+            let first = self.ptr as usize;
+            let base = self.data.0.as_ptr() as usize;
+            let index = (first - base) / size;
+            debug_assert_eq!((first - base) % size, 0);
+            // Remove the element at that index and return it.
+            self.data.0.remove(index)
```

Comment:
Gnarly but it works fine. Maybe a comment to explain that the len of the Vec might be bigger than the array's current len

Reply:
Okay, I've added a explanatory comment. If someone changes the implementation in the future, the test will also help catch this edge case.